modules/template_architecture_definition/outputs.tf

```diff
@@ -1,3 +1,126 @@
 output "architecture_definition_json" {
   value = local.has_architecture_definition_override ? data.local_file.architecture_definition_override_json[0].content : local.template_file
+
+  precondition {
+    condition     = length(local.management_group_configuration) != 0
+    error_message = "The management group configuration is required"
+  }
+
+  precondition {
```

Comment:
Could this be done in a `for` loop based on the `keys()`? To remove the duplicate code.